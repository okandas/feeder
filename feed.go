package feeder

// Feeds are multiple feeds within the application
type Feeds struct {
	Feeds []Feed `json:"feeds"`
}

// Feed represents a feed
type Feed struct {
	Name          string   `json:"name"`
	Size          int      `json:"max_size"` // items can be in the feed
	PerPage       int      `json:"per_page"`
	Activities    []*Event `json:"activities"`
	P	  		  Backend `json:"-"`
}

// NewFeed instantiates a new feed struct
func NewFeed(name string, size, perPage int,  b Backend) *Feed {
	return &Feed{
		Name:          name,
		Size:          size,
		PerPage:       perPage,
		Activities:    []*Event{},
		P: 	   b,
	}

}

//TODO: should also register our feed in the register and then have a way of accessing that feed with a method
