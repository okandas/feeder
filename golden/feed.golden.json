{
    "name": "notifications",
    "max_size": 5,
    "per_page": 2,
    "activities": [ 
        { 
            "user_id": "okandas",
            "last_read": "2018-03-22T10:00:00-05:00",
            "unread": 2,
            "count": 3,
            "activity": [
                {
                    "value": "the gods have removed",
                    "at": "2018-03-22T10:00:00-05:00",
                    "method": "crowdsourced"
                },
                {
                    "value": "watched formula one",
                    "at": "2018-03-21T10:00:00-05:00",
                    "method": "import"
                },
                {
                    "value": "secure the bag",
                    "at": "2018-03-23T10:00:00-05:00",
                    "method": "single supplier"
                }
            ]
        }
    ]

}